### PR TITLE
Implement Role-based authorization UI for Viewer and Editor roles

### PR DESCRIPTION
**Pull request by jaredscheib**
- [x] Rebased/mergable
  - [x] Tests pass

Connect #1916 
Connect #1913 
Connect #2146 

This also hides the Admin tab from Viewer and Editor.

## To test:
1. run `git cherry-pick -n fc3e4dc8` to force your Role to be Viewer
1. build the server
1. run server with auth, including oauth client and secret

### CODE REVIEW

File: ui/src/style/components/source-indicator.scss [hunk 1]
```diff
@@ -1,7 +1,8 @@
 /*
-    Source Indicator component styles
-    ----------------------------------------------------------------
+    Source & Role Indicator component styles
+    ----------------------------------------------------------------------------
 */
+.role-indicator,
```
Comment (lukevmorris) [inline ui/src/style/components/source-indicator.scss]: The name of this file is no longer accurate

File: ui/src/auth/Authorized.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    case ADMIN_ROLE:
+      return meRole === ADMIN_ROLE || meRole === SUPERADMIN_ROLE
+    case SUPERADMIN_ROLE:
+      return meRole === SUPERADMIN_ROLE
+    default:
+      return false
+  }
+}
+
+const getRoleName = ({roles: [{name}, ..._]}) => name
```
Comment (lukevmorris) [inline ui/src/auth/Authorized.js]: This is duplicated in RoleIndicator

File: ui/src/auth/Authorized.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  } else if (
+    isUsingAuth &&
+    !isUserAuthorized(meRole, requiredRole) &&
+    propsOverride
+  ) {
+    return React.cloneElement(
+      React.isValidElement(children) ? children : children[0],
+      {...additionalProps, ...propsOverride}
+    )
+  }
```
Comment (lukevmorris) [inline ui/src/auth/Authorized.js]: See me after class.

File: ui/src/auth/Authorized.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+  return replaceWith ? replaceWith : null
```
Comment (lukevmorris) [inline ui/src/auth/Authorized.js]: `return replaceWith || null`

File: ui/src/auth/Authorized.js (hunk 2 above)
Comment (jaredscheib) [reply ui/src/auth/Authorized.js]: 👍

File: ui/src/style/components/source-indicator.scss (hunk 1 above)
Comment (jaredscheib) [reply ui/src/style/components/source-indicator.scss]: 👍 changed

**Pull request merged**